### PR TITLE
VMware: new module : vmware_cluster_facts

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_cluster_facts.py
test/integration/targets/vmware_cluster_facts/aliases
test/integration/targets/vmware_cluster_facts/tasks/main.yml

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_cluster_facts.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                self.module.fail_json(msg="Failed to find datacenter '%s'" % datacenter)
+            self.cluster_objs = self.get_all_cluster_objs(parent=datacenter_obj)
+        elif cluster_name:
+            cluster_obj = find_cluster_by_name(self.content, cluster_name=cluster_name)
+            if cluster_obj is None:
+                self.module.fail_json(msg="Failed to find cluster '%s'" % cluster_name)
+
+            self.cluster_objs = [cluster_obj]
+
+    def get_all_cluster_objs(self, parent):
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_cluster_facts.py:124]: Could potentially be moved to the module_utils? (same for lines 112 -> 122)

**Review by pdellaert — APPROVED** [review]
LGTM
Small comment if these methods could be useful for other modules in the future?

**Comment by Akasurde** [comment]
rebuild_merge

**Comment by Akasurde** [comment]
@pdellaert Thanks for review.

**Pull request merged**